src/PMGDQueryHandler.cc

```diff
@@ -241,6 +244,50 @@ int PMGDQueryHandler::add_node(const protobufs::AddNode &cn,
     return 0;
 }
 
+int PMGDQueryHandler::update_node(const protobufs::UpdateNode &un,
+                    protobufs::CommandResponse *response)
+{
+    long id = un.identifier();
+    bool query = un.has_query_node();
+
+    // Make sure there is either an id or a query node
+    if (id < 0 && !query) {
+        set_response(response, PMGDCmdResponse::Error, "No way to find update node\n");
+        return -1;
+    }
+
+    auto it = _cached_nodes.find(id);
```

Comment:
Shouldn't this check whether id >= 0 before calling find? (See suggestion below.)

Reply:
Fixed